diff --git a/mountall.sh b/mountall.sh
index 94bae42..44fafec 100755
--- a/mountall.sh
+++ b/mountall.sh
@@ -11,6 +11,14 @@
 
 . /etc/default/rcS
 
+ROOT_IS_ON_NAND=`cat /proc/cmdline | grep "root=ubi0"`
+if [ "$ROOT_IS_ON_NAND" ]; then
+	mkdir -p /home/root/app
+	ubiattach /dev/ubi_ctrl -m 3 -d1
+	mount -t ubifs -o defaults,noatime,rw ubi1:app /home/root/app
+	echo "booted from nand"
+fi
+
 #
 # Mount local filesystems in /etc/fstab. For some reason, people
 # might want to mount "proc" several times, and mount -v complains
